### ISSUE TITLE
Minor enhancement: Admin UI missing lan's

### ISSUE THREAD

**Issue by rica-carv** [e107inc/e107#1770]
I'm curently translating a site i'm setting up for some languages, and on the admin UI i've got my default language defined as language for the admin UI area.
I've got a pair of plugins written on my own, also on current development.
My proposal is that, for instance, if i haven't got yet the language files for that plugin present in their location, tha language lans default to what i've defined on the preferences.
Currently, if a language is not present, it just dumps the LAN constant names on the admin UI, but if i got the files on the language folder, it displays the default as i've set up on preferences....
Can't this last option covers both situations, when language file are present and when they're not?

**Comment by tgtje** [comment]
Imo opinion no, since the system uses the fallback to English iff any... 
That's why English language files are dominating (and always should be present). The LAN's are basically just the (let's call them) triggers for fallback..(since no determination of native language files (defines) are made, it reverts to the 'coding ' instead.. So working state will always be provided, just not the translation.

**Comment by rica-carv** [comment]
@tgtje Maybe i expressed myself wrong. I don't mind if the fallback is english, or any other language, it should always fallback even if a language LAN file is missing, not display the LAN constant string, if you get my point....
A further enanhecement would be a fallback this way: selected LAN files not present -> select default LAN for admin UI.
If not present,  -> Default to english.....

**Comment by tgtje** [comment]
No, you did not express yourself wrong, and i understand your thinking, but the display of the LAN constant is the piece that is occupied inside the file that uses it. 
Unless the default (English lan + define + native) is not supplied your stuck! 

I do not see an easy way to hide/replace the LAN constant (= piece of code), and represent something else instead. _That it does show_ is also a reminder that there is something not ok with language....
The only solution that comes close would be hardcode the languagetext (aka define) straight in file. (which in itself takes away the advantage of using multi-language).

**Comment by rica-carv** [comment]
@tgtje Once again, i might express my ideia wrong.
I got the english default core languange installed, but if i switch to another language, for instance, portuguese, and if the current plugin i'm working currently does not have that portuguese language set up (but does have the english one!), in the admin UI area (and also on the frontend!), what's displayed is the LAN_XXXX name constants, not the default english language string!
So, what i asked is, if missing any language file other than the default one (ie, english), LAN_XXX constants shouldn't be displayed, but the corresponding string defined by the default language...

**Comment by tgtje** [comment]
Got it, needs probably a if else statement when calling the define like if i can not find Spanish than use English.
Hmm. personally thinking iff the old piece does not what you would like to see (stepping away from the new use of language call e107::lan (for time being, because going forward and not back..another solution would be needed..)> 

`include_lan(e_PLUGIN."PLUGINNAME/languages/".e_LANGUAGE.".php");` 
and does need at least the provided English.php file ( file inside main lang folder, not in English/Spain etc.. ) Small test does provide the 'wish' , but is as 'old method' , so new method needs enhancement/changes..
